lib/types/local-time.js

```diff
@@ -27,257 +29,229 @@ const nanoSecInMillis = Long.fromNumber(1000000);
-}
-
-/**
- * Parses an string representation and returns a new LocalDate.
- * @param {String} value
- * @returns {LocalTime}
- */
-LocalTime.fromString = function (value) {
-    if (typeof value !== "string") {
-        throw new Error("Argument type invalid: " + util.inspect(value));
+    get hour() {
+        return this.internal.hour;
     }
-    const parts = value.split(":");
-    let millis =
-        parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
-    let nanos;
-    if (parts.length === 3) {
-        const secParts = parts[2].split(".");
-        millis += parseInt(secParts[0], 10) * 1000;
-        if (secParts.length === 2) {
-            nanos = secParts[1];
-            //add zeros at the end
-            nanos = nanos + utils.stringRepeat("0", 9 - nanos.length);
-        }
+
+    set hour(newValue) {
+        throw new Error("Hour length is read-only");
     }
-    return LocalTime.fromMilliseconds(millis, parseInt(nanos, 10) || 0);
-};
 
-/**
- * Uses the current local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
- * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
- * @returns {LocalTime}
- */
-LocalTime.now = function (nanoseconds) {
-    return LocalTime.fromDate(new Date(), nanoseconds);
-};
+    /**
+     * Parses an string representation and returns a new LocalDate.
+     * @param {String} value
+     * @returns {LocalTime}
+     */
+    static fromString(value) {
+        if (typeof value !== "string") {
+            throw new TypeError(
+                "Argument type invalid: " + util.inspect(value),
+            );
+        }
 
-/**
- * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
- * @param {Date} date Local date portion to extract the time passed since midnight.
- * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the nanosecond time portion.
- * @returns {LocalTime}
- */
-LocalTime.fromDate = function (date, nanoseconds) {
-    if (!util.isDate(date)) {
-        throw new Error("Not a valid date");
+        const parts = value.split(":");
+        let millis =
+            parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
+        let nanos;
+        if (parts.length === 3) {
+            const secParts = parts[2].split(".");
+            millis += parseInt(secParts[0], 10) * 1000;
+            if (secParts.length === 2) {
+                nanos = secParts[1];
+                // add zeros at the end
+                nanos = nanos + utils.stringRepeat("0", 9 - nanos.length);
+            }
+        }
+        return LocalTime.fromMilliseconds(millis, parseInt(nanos, 10) || 0);
     }
-    //Use the local representation, only the milliseconds portion
-    const millis =
-        (date.getTime() + date.getTimezoneOffset() * -60000) % millisInDay;
-    return LocalTime.fromMilliseconds(millis, nanoseconds);
-};
 
```

Comment:
Maybe considering checking (by regex?) if format is correct?
Currently this will throw uninformative errors if wrong format is suplied.